### PR TITLE
[Torchscript] Enable inference optimizations on the scripted model

### PR DESCRIPTION
**Pull request by pooyanamini**
**Patch description**
Enable inference optimization on the torchscripted model. This also
fixes a crash in inference from the torchscripted model when we had
multi-line context.

**Testing steps**
Tried scripting a model and running inference on it for multi-turn context and made sure we don't see the crash and the output is correct.
buck run //deeplearning/projects/parlai:parlaicmd -- torchscript --model-file manifold://cair_models/tree/cairaoke/experimental/christopherhlin/[f335033955](https://www.internalfb.com/intern/fblearner/details/335033955)/cstudio_reminder_r4 --model fb:bart/cairaoke_bart --scripted-model-file manifold://cair_models/tree/cairaoke/experimental/test/script_reminder_debug.pt --no_cuda --input 'what are my reminders for tomorrow|api_resp: get_entity.date = 2021-07-12T00:00:00.000-07:00'

**Other information**

### CODE REVIEW

**Comment by pooyanamini** [comment]
I tried reproducing the error by running this command:
```
buck run //deeplearning/projects/parlai:parlaicmd \
-- torchscript \
--model bart \
--scripted-model-file manifold://cair_models/tree/cairaoke/experimental/test/script_debug.pt \
--no_cuda \
--input "Don't have a cow, man!"
```
but the command above worked. I will try to see if I can reproduce it and debug it. If not, I will add a flag as you suggested

**Comment by stephenroller** [comment]
Can we check which version of PyTorch that inference-fast thing was introduced in?

**Comment by stephenroller** [comment]
Also I don't know about --no-cuda

File: parlai/scripts/torchscript.py [hunk 1]
```diff
@@ -54,17 +54,22 @@ def export_model(opt: Opt):
     instantiated = script_class(agent)
     if not opt["no_cuda"]:
         instantiated = instantiated.cuda()
-    jit_model = torch.jit.optimize_for_inference(torch.jit.script(instantiated.eval()))
+    scripted_model = torch.jit.script(instantiated.eval())
+    if opt.get("enable-inference-optimizations"):
```
Comment (stephenroller) [inline parlai/scripts/torchscript.py:57]: should be underscores

File: parlai/scripts/torchscript.py [hunk 2]
```diff
@@ -96,7 +96,7 @@ def setup_args() -> ParlaiParser:
     )
     parser.add_argument(
         "-eio",
-        "--enable-inference-optimization",
+        "--enable_inference_optimization",
```
Comment (stephenroller) [inline parlai/scripts/torchscript.py]: but this should be dashes :P

Comment (pooyanamini) [reply parlai/scripts/torchscript.py:101]: I can make it use dashes for consistency with the other parameters. Let me know which one you prefer

**Comment by pooyanamini** [comment]
@stephenroller these failures seem to be unrelated to the PR. can you please check if any other changes are needed to the PR?

**Comment by pooyanamini** [comment]
@stephenroller , @EricMichaelSmith it seems that finally the CI checks are passing. Can you please take a look?

**Review by stephenroller — APPROVED** [review]

**Pull request merged**